src/cinder/linux/GstPlayer.cpp

```diff
@@ -20,1298 +17,1147 @@
+        GError* err = nullptr;
+        gchar* uri = gst_filename_to_uri( path.c_str(), &err );
+        uriPath = std::string( static_cast<const char*>( uri ) );
+        g_free( uri );
+        if( err ) { 
+            g_free( err );
+        }
+        mGstData.isStream = false;
+    }
+    else {
+        mGstData.isStream = true;
+    }
 
-void GstPlayer::unlinkAudioBranch() 
-{
-	// If we had audio remove and unlink the audio elements.
-	// On the next load the audio branch will be re-linked if an audio stream is present.
-	if( mGstPipeline && mGstData.mHasAudio ) {
-		gst_element_set_state( mGstData.mAudioQueue, GST_STATE_NULL );
-		gst_element_get_state( mGstData.mAudioQueue, nullptr, nullptr, GST_CLOCK_TIME_NONE );
-		gst_element_set_state( mGstData.mAudioconvert, GST_STATE_NULL );
-		gst_element_get_state( mGstData.mAudioconvert, nullptr, nullptr, GST_CLOCK_TIME_NONE );
-		gst_element_set_state( mGstData.mAudiosink, GST_STATE_NULL );
-		gst_element_get_state( mGstData.mAudiosink, nullptr, nullptr, GST_CLOCK_TIME_NONE );
+    // set the new movie path
+    const gchar* uri = ( ! uriPath.empty() ) ? uriPath.c_str() : path.c_str();
+    g_object_set( G_OBJECT ( mGstData.pipeline ), "uri", uri, nullptr );
 
-		gst_bin_remove_many( GST_BIN( mGstPipeline ), mGstData.mAudioQueue, mGstData.mAudioconvert, mGstData.mAudiosink, nullptr );
-		mGstData.mHasAudio = false;
-	}
+    // and preroll.
+    setPipelineState( GST_STATE_PAUSED );
 }
 
 void GstPlayer::play()
 {
-	setPipelineState( GST_STATE_PLAYING );
+    setPipelineState( GST_STATE_PLAYING );
 }
 
 void GstPlayer::stop()
 {
-	setPipelineState( GST_STATE_PAUSED );   
+    setPipelineState( GST_STATE_PAUSED );   
 }
 	
 bool GstPlayer::newVideo() const
 {
-	return mGstData.mVideoHasChanged;
+    return mGstData.videoHasChanged;
 }
 
 int GstPlayer::width() const
 {
-	return mGstData.mWidth;
+    return mGstData.width;
 }
 
 int GstPlayer::height() const
 {
-	return mGstData.mHeight;
+    return mGstData.height;
 }
 
 bool GstPlayer::isPaused() const
 {
-	return mGstData.mPaused;
+    return mGstData.isPaused;
 }
 
 bool GstPlayer::isPlayable() const
 {
-	return mGstData.mIsPlayable;
+    return mGstData.isPlayable;
 }
 
 bool GstPlayer::isLoaded() const
 {
-	return mGstData.mIsLoaded;
+    return mGstData.isLoaded;
 }
 
 bool GstPlayer::isLiveSource() const
 {
-	return mGstData.mIsLive;
+    return mGstData.isLive;
 }
 
 bool GstPlayer::isBuffering() const
 {
-	return mGstData.mIsBuffering;
+    return mGstData.isBuffering;
 }
 
 bool GstPlayer::isPrerolled() const
 {
-	return mGstData.mIsPrerolled;
+    return mGstData.isPrerolled;
 }
 
 GstVideoFormat GstPlayer::format() const
 {
-	return mGstData.mVideoFormat;
+    return mGstData.videoFormat;
 }
 
 float GstPlayer::getFramerate() const
 {
-    return mGstData.mFrameRate;
+    return mGstData.frameRate;
 }
 
 bool GstPlayer::hasAudio() const
 {
-    return mGstData.mHasAudio;
+    if( ! mGstData.pipeline ) return false;

```

Comment:
Can you place the return statements on a new line (there are a few below as well)? It makes stepping through the code easier.